apis/kueue/v1alpha1/tas_types.go

```diff
@@ -27,3 +53,57 @@ const (
 	// The label is set when starting the Job, and removed on stopping the Job.
 	PodSetLabel = "kueue.x-k8s.io/podset"
 )
+
+// TopologySpec defines the desired state of Topology
+type TopologySpec struct {
+	// TopologyLevels define the levels of topology.
+	//
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	Levels []TopologyLevel `json:"levels,omitempty"`
+}
+
+// TopologyLevel defines the desired state of TopologyLevel
+type TopologyLevel struct {
+	// NodeLabel indicates the name of the node label for a specific topology
+	// level.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=316
+	// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+	NodeLabel string `json:"nodeLabel,omitempty"`
+}
+
+// TopologyStatus defines the observed state of Topology
+type TopologyStatus struct {
+}
+
+// +kubebuilder:object:root=true
```

Comment:
```suggestion
// +genclient
// +genclient:nonNamespaced
// +kubebuilder:object:root=true
```

Let's generate client as well.

Reply:
Added